### PR TITLE
refactor: Code cleanup (AB#18016)

### PR DESCRIPTION
**Pull request by DmytroLohvynenko**
Fixes [AB#18119](https://dev.azure.com/CluedIn-io/c054b4ae-1dab-43c2-af97-3683c744782f/_workitems/edit/18119)
Remove override of VerifyConnection that copy base implementation

Fixes [AB#18120](https://dev.azure.com/CluedIn-io/c054b4ae-1dab-43c2-af97-3683c744782f/_workitems/edit/18120)
Remove unnecessary DI registrations in AzureServiceBusConnectorComponent

Fixes [AB#18157](https://dev.azure.com/CluedIn-io/c054b4ae-1dab-43c2-af97-3683c744782f/_workitems/edit/18157)
Remove SQL tests

### CODE REVIEW

**Comment by devbryan0225** [comment]
PR naming convention:
https://www.conventionalcommits.org/en/v1.0.0/

Add `(AB#18016)` to the title in this PR. ex. `feat: Cleanup code (AB#18016)`

It will then tie your PR to your [PBI 18016](https://dev.azure.com/CluedIn-io/CluedIn/_workitems/edit/18016)

image

More on PRs:
https://dev.azure.com/CluedIn-io/CluedIn/_wiki/wikis/CluedIn.wiki/77/Pull-Request-Process

File: src/Connector.SqlServer/Connector/AzureServiceBusConnector.cs [hunk 1]
```diff
@@ -62,26 +62,6 @@ public override async Task CreateContainer(ExecutionContext executionContext, Gu
...
-        public override async Task RenameContainer(ExecutionContext executionContext, Guid providerDefinitionId, string id, string newName)
-        {
-            await Task.FromResult(0);
-        }
-
-        public override async Task RemoveContainer(ExecutionContext executionContext, Guid providerDefinitionId, string id)
-        {
-            await Task.FromResult(0);
-        }
-
```
Comment (devbryan0225) [inline src/Connector.SqlServer/Connector/AzureServiceBusConnector.cs:84]: better to leave them where they are for ease of audit trail purposes, if they're only moved around.

File: src/Connector.SqlServer/Connector/AzureServiceBusConnector.cs [hunk 2]
```diff
@@ -104,23 +84,6 @@ public override async Task<string> GetValidContainerName(ExecutionContext execut
...
-            return await Task.FromResult(new List<IConnectionDataType>());
-        }
-
-        public override async Task<bool> VerifyConnection(ExecutionContext executionContext, Guid providerDefinitionId)
-        {
-            var config = await GetAuthenticationDetails(executionContext, providerDefinitionId);
-
-            return await VerifyConnection(executionContext, config.Authentication);
-        }
-
```
Comment (devbryan0225) [inline src/Connector.SqlServer/Connector/AzureServiceBusConnector.cs:123]: better to leave them where they are for ease of audit trail purposes, if they're only moved around.

**Review by devbryan0225 — COMMENTED** [review]
please add summary of what is done here to the PR description for easier tracking

File: src/Connector.SqlServer/Connector/AzureServiceBusConnector.cs (hunk 1 above)
Comment (DmytroLohvynenko) [reply src/Connector.SqlServer/Connector/AzureServiceBusConnector.cs:84]: My idea was to move them bellow all real code and hide in `#region NotImplementedMembers `, so developers would spend less time scrolling through them up and down when working with this class.

File: src/Connector.SqlServer/Connector/AzureServiceBusConnector.cs (hunk 2 above)
Comment (DmytroLohvynenko) [reply src/Connector.SqlServer/Connector/AzureServiceBusConnector.cs]: +

File: Nuget.config [hunk 3]
```diff
@@ -11,5 +11,6 @@
         <clear/>
         <add key="nuget.org" value="https://api.nuget.org/v3/index.json" />
         <add key="Public" value="https://pkgs.dev.azure.com/CluedIn-io/Public/_packaging/Public/nuget/v3/index.json" />
+        <add key="release" value="https://pkgs.dev.azure.com/CluedIn-io/_packaging/release/nuget/v3/index.json" />
```
Comment (andrew-hardy) [inline Nuget.config:14]: Revert the changes to this file as CluedIn.Connector.AzureServiceBus is a public repo and the release nuget feed is not public

File: Packages.props [hunk 4]
```diff
@@ -45,6 +45,7 @@
     <PackageReference Update="Dapper" Version="2.0.35" />
     <PackageReference Update="Dapper.SqlBuilder" Version="2.0.35" />
     <PackageReference Update="Microsoft.AspNet.WebApi.Client" Version="5.2.7" />
+    <PackageReference Update="CluedIn.Connector.Common" Version="3.4.3" />
```
Comment (andrew-hardy) [inline Packages.props:48]: 3.4.3 is not available on the public feed

**Review by andrew-hardy — CHANGES_REQUESTED** [review]
In addition, please add a test rather than delete the test scaffolding

**Comment by DmytroLohvynenko** [comment]
> In addition, please add a test rather than delete the test scaffolding

I'm starting on project soon, not sure how much free time I will get, so at least I would like to finish with cleaning this repo.

File: Packages.props (hunk 4 above)
Comment (DmytroLohvynenko) [reply Packages.props:48]: how to add it there?

Comment (andrew-hardy) [reply Packages.props:48]: release needs to be merged into master. I don't know if the release branch is ready for that though.

File: Nuget.config (hunk 3 above)
Comment (DmytroLohvynenko) [reply Nuget.config]: ++

File: Packages.props (hunk 4 above)
Comment (DmytroLohvynenko) [reply Packages.props]: Good suggestion, changed to 3.4.0.

For some weird reasons I wasn't able to check that `Connector.Common` available in Public.
If I click `release` for example, I could see package and versions.
But if I click `Public` - just white window.
image
image
image

**Comment by andrew-hardy** [comment]
Test scaffolding is still deleted. Please refer to my previous comment about adding a test rather than delete the test scaffolding.

**Comment by DmytroLohvynenko** [comment]
> Test scaffolding is still deleted. Please refer to my previous comment about adding a test rather than delete the test scaffolding.


@andrew-hardy sorry but the work in question was about cleaning code in repository. Adding new features or tests were not in scope of related tasks.

Do I need to restore scaffolding (actually copy-pasted SQL Connector tests with disabled code, that always return false positives) to move on?

**Comment by andrew-hardy** [comment]
I don't agree with referencing `CluedIn.Connector.Common`. `CluedIn.Connector.Common` contains breaking changes in every revision from 3.3 -> 3.4 -> 3.5 . It has been referenced in other connectors such as `CluedIn.Connector.AzureDataLake` and `Cluedin.Connector.AzureDedicatedSqlPool`, with the end result being that these connectors cannot run side by side with `CluedIn.Connector.SqlServer 3.3` (the current default in the AMA). 

I suggest closing this PR. Perhaps in the future if a backwards compatible `CluedIn.Connector.Common` is produced and backwards compatibility checks are performed on build it could be considered a safe dependency.

**Review by andrew-hardy — APPROVED** [review]

**Pull request merged**